src/include/main/client_context.h

```diff
@@ -73,6 +78,8 @@ class ClientContext {
 
     void setExtensionOption(std::string name, common::Value value);
 
+    inline common::RandomEngine& getRandomEngine() { return *randomEngine; }
```

Comment:
I think we just return pointer. If you return a reference and `RandomEngine` implements a default copy constructor, someone might trigger copy unexpected through `auto engine = context.getRandomEngine()`